[Sponsor Page] Sponsors Tier cleanup

CarsonKiibi opened the pull request:
-fixed stuff based on comments from last pr -> reformatted to move stuff out of component that isn't based on state, deleted useless styling classes, etc 
-cleaned up code overall
-left/right navigation buttons and bottom nav bar now stay in the same place no matter the content size
-fixed some spacing stuff, should be more responsive overall

The only issues with the component now is the transition still doesn't happen when clicking the navbar icons (only the left/right arrows cause transitions), and I hardcoded DNA strand/card height to allow the button placements to be consistent. I don't really know how to explain it but lines 144 + 173 in sponsors_tier.tsx feel wrong. But I think we could just hardcore better based on screen size.

Commit:
-added sponsors_tier TODO: fix transitions, card height

Commit:
-added static height TODO transition and button position

daniel-panhead committed:
Delete extraneous PastSponsors.stories.ts

daniel-panhead committed:
Get rid of default styles

daniel-panhead committed:
Merge branch 'daniel/cleanup' into Carson

daniel-panhead committed:
Remove unnecessary styles now that global styles are reset

Commit:
Changed file locations so tailwind styling works. Adjusted to be mainly tailwind styling. Changed button location + added some other basic styling to clean up.

Commit:
Changed styling to tailwind and modified button location + some other simple styling

Commit:
-fixed stuff that should be static

IcedTea2K committed:
Merge branch 'main' into Carson

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+import React, { useState, useEffect } from "react";
```
Would suggest moving this file into sponsors folder. Another thing to nitpick about, use Pascal naming conventions for the file as well (similar to how you'd name the component). E.g., `SponsorsTier.tsx`

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+import Image from "next/image";
+
+import "./sponsors_tier.css";
+import { on } from "events";
```
remove unused imports

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+import React, { useState, useEffect } from "react";
+import Image from "next/image";
```
remove unused imports

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+    link: "https://google.com",
+  },
+  {
+    image: "./gold_img1.jpg",
```
create an assets folder in src, and put the image in there. And import them as follows
```tsx
import  googleIcon from "@/assets/sponsors/googleIcond.jpg"

...
{
   image: googleIcon
   header: ...
   subheader: ...
   text: ...
   link: ....
}
...
```

Also, do you mind using actual images of the logos? It's gonna be easier to visualize them.

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+  };
+
+  const { image, header, subheader, text, link } =
+    currentContentData[currentContent];
```
LIke I mentioned above, use actual logos instead of placeholder.

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+
+      {/* Right Content */}
+      <div className="flex w-full sm:px-5 items-center  my-auto">
+        <div className="p-4 flex flex-col h-full w-full sm:p-7">
```
`w-full` here might look quite silly when we put this onto the page. It might stretch out the entire page. 

But for now, this could be kept. We need to sort out the layout for the page first (i.e., how wide, what margin).

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+            >
+              &lt;
+            </button>
+            <div
+              className={`flex flex-col ${
+                transitionDirection === "left" ? "cardTransitionLeft" : ""
+              }
+    ${transitionDirection === "right" ? "cardTransitionRight" : ""}
+    ${transitionDirection === "newCard" ? "newCardTransition" : ""}`}
+              onAnimationEnd={handleTransitionEnd}
+            >
+              <div className="flex flex-row items-center">
+                <div className="logo-temp"></div>
+                <h2 className="text-2xl font-bold black">{header}</h2>
+              </div>
+              <h3 className="text-lg subheader">{subheader}</h3>
+              <p className="black">{text}</p>
+              <a
+                className="link explore-more"
+                href={link}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                Explore More
+              </a>
+            </div>
```
I'd recommend abstract this into another component `SponsorCard` maybe. But you don't really have to, since you'll own this code and will be in charge of maintaining it.

(please don't quit as well 😭 )

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+      <div className="flex w-full sm:px-5 items-center  my-auto">
+        <div className="p-4 flex flex-col h-full w-full sm:p-7">
+          <div className="flex flex-row items-center justify-between h-[600px] sm:h-[500px] md:h-[350px]">
+            <button
+              className="nav-button sm:mx-10 mr-3"
+              onClick={() => handleNavigation("prev")}
+            >
+              &lt;
+            </button>
```
Please use icon button that comes with pre-line (for standardized style)
https://preline.co/docs/buttons.html#icon

IcedTea2K commented on src/components/sponsors_tier.tsx:
```diff
@@ -0,0 +1,231 @@
+                Explore More
+              </a>
+            </div>
+            <button
+              className="nav-button sm:mx-10 ml-3"
+              onClick={() => handleNavigation("next")}
+            >
+              &gt;
+            </button>
```
Similarly, use pre-line icon button for this.

IcedTea2K commented on src/components/sponsors_tier.css:
```diff
@@ -0,0 +1,112 @@
+.contentDataPlaceholder {
```
Loved the animation. But preferably not to use separate css. Could try using Preline animation for the carousel. I had a quick look and it looks pretty simple. You can even do the infinite-loop for automatic scrolling as well! https://preline.co/plugins/html/carousel.html

IcedTea2K commented on src/stories/assets/img1.jpg:
what da heck Carson ahahahahahhaa??? Do remove if you're not using it though. Same thing for img2.jpg

IcedTea2K commented on src/stories/sponsors_tier.stories.ts:
Move this into the sponsors folder within `stories/sponsors` for organizational sake. Also rename the file to `SponsorsTier.stories.ts`

IcedTea2K commented on src/stories/sponsors_tier.stories.ts:
```diff
@@ -0,0 +1,29 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { within, userEvent } from "@storybook/testing-library";
```
remove unused

IcedTea2K commented on src/stories/sponsors_tier.stories.ts:
```diff
@@ -0,0 +1,29 @@
+
+import { SponsorsTier } from "../components/sponsors_tier";
+
+// const meta = {
+//   title: "Example/Contact_Hero_Header",
+//   component: Contact_Hero_Header,
+//   parameters: {
+//     // More on how to position stories at: https://storybook.js.org/docs/react/configure/story-layout
+//     layout: "fullscreen",
+//   },
+// } satisfies Meta<typeof Contact_Hero_Header>;
+
```
remove unwanted code

IcedTea2K commented on src/stories/sponsors_tier.stories.ts:
```diff
@@ -0,0 +1,29 @@
+  component: SponsorsTier,
+  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
+  tags: ["autodocs"],
+  parameters: {
+    // More on how to position stories at: https://storybook.js.org/docs/react/configure/story-layout
+    layout: "fullscreen",
+  },
```
can you add `viewport` to allow us to see the mobile view as well? check out this link https://storybook.js.org/docs/essentials/viewport!

IcedTea2K reviewed (changes requested):
Good stuff!!! I loved the animation. There are things you could cleanup and organize such as filenames and structures, but overall really good. For the animation, do look into Preline animations `https://preline.co/plugins/html/carousel.html`, they are pretty neat.

Commit:
src->components->sponsors->SponsorsTier.tsx line 195 need to change to tailwind

-Added sponsor logos instead of placeholder boxes
-cleaned up (other than css file cuz thats what needs to be fixed as described above

Commit:
Merge branch 'Carson' of https://github.com/ubc-biomod/biomod-website into Carson

Commit:
-changed contentItem to camel case
-removed some useless comments
-changed arrow function to simple function
-changed <img> to <Image> (next image)

CarsonKiibi committed:
Delete src/components/joinus/JoinUsHero.tsx

-accidently added this old file

CarsonKiibi committed:
Delete src/stories/sponsors/JoinUsHero.stories.ts

-accidently pushed this + the component whoops

daniel-panhead committed:
Rework animation for sponsor tier with framer motion

daniel-panhead committed:
Populate sponsor tier sponsors

daniel-panhead committed:
Add interactive DNA graphic

daniel-panhead commented:
![image](https://github.com/user-attachments/assets/5be21dbd-75bc-42d2-99f4-7887b02b841c)

my face hurts i want to sleep

daniel-panhead commented:
![image](https://github.com/user-attachments/assets/5be21dbd-75bc-42d2-99f4-7887b02b841c)

my face hurts i want to sleep

daniel-panhead committed:
Add hover scale effect to dna buttons

IcedTea2K committed:
remove extra css file

IcedTea2K reviewed (approved):
LGTM 👍

daniel-panhead committed:
Merge remote-tracking branch 'origin/main' into Carson
